vision/getting-started/image_segmentation.ipynb

```diff
@@ -0,0 +1,1003 @@
+        "                y1 = min(y1, y)\n",
+        "                x2 = max(x2, x)\n",
+        "                y2 = max(y2, y)\n",
+        "\n",
+        "    if x1 > x2 or y1 > y2:\n",
+        "        return None  # No masked area found\n",
+        "    else:\n",
+        "        return (x1, y1, x2 + 1, y2 + 1)  # Add 1 to include the last pixel\n",
+        "\n",
+        "\n",
+        "def get_labeled_boxes(response: PredictResponse) -> list:\n",
+        "    labeled_boxes = []\n",
+        "    for prediction in response.predictions:\n",
+        "        mask_pil = prediction_to_mask_pil(prediction)\n",
+        "        bounding_box = get_bounding_box(mask_pil)\n",
+        "        if bounding_box:\n",
+        "            label, score = get_prediction_top_label(prediction)\n",
+        "            score = round(float(score), 3)\n",
+        "            labeled_box = (label, score, bounding_box)\n",
+        "            labeled_boxes.append(labeled_box)\n",
+        "\n",
+        "    return labeled_boxes\n",
+        "\n",
+        "\n",
+        "# Constructs a Vertex AI PredictRequest and uses it to call Image Segmentation.\n",
+        "def call_vertex_image_segmentation(\n",
+        "    image_bytes=None,\n",
+        "    gcs_uri=None,\n",
+        "    mime_type=None,\n",
+        "    mode=\"foreground\",\n",
+        "    prompt=None,\n",
+        "    scribble_bytes=None,\n",
+        "    mask_dilation=None,\n",
+        "    max_predictions=None,\n",
+        "    confidence_threshold=None,\n",
+        "):\n",
+        "    instances = []\n",
+        "\n",
+        "    if image_bytes:\n",
+        "        instances.append(\n",
+        "            {\n",
+        "                \"image\": {\n",
+        "                    \"bytesBase64Encoded\": image_bytes,\n",
+        "                    \"mimeType\": mime_type,\n",
+        "                },\n",
+        "            }\n",
+        "        )\n",
+        "    elif gcs_uri:\n",
+        "        instances.append(\n",
+        "            {\n",
+        "                \"image\": {\"gcsUri\": gcs_uri},\n",
+        "            }\n",
+        "        )\n",
+        "\n",
+        "    if scribble_bytes:\n",
+        "        instances[0][\"scribble\"] = {\n",
+        "            \"image\": {\n",
+        "                \"bytesBase64Encoded\": scribble_bytes,\n",
+        "                \"mimeType\": \"image/png\",\n",
+        "            },\n",
+        "        }\n",
+        "\n",
+        "    if prompt:\n",
+        "        instances[0][\"prompt\"] = prompt\n",
+        "\n",
+        "    parameters = {\"mode\": mode}\n",
+        "\n",
+        "    if mask_dilation:\n",
+        "        parameters[\"maskDilation\"] = mask_dilation\n",
+        "    if max_predictions:\n",
+        "        parameters[\"maxPredictions\"] = max_predictions\n",
+        "    if confidence_threshold:\n",
```

Comment:
This method should be simplified if you use the Vertex AI Python SDK

Reply:
SDK not available for this yet.